include/http/request_handler.js

```diff
@@ -144,6 +145,87 @@ module.exports = function RequestHandlerModule(pb) {
 
     /**
      * @static
+     * @method saveRedirectHost
+     * @param {String} hostname
+     * @param {String} oldHostname
+     * @param {String} uid
+     */
+    RequestHandler.saveRedirectHost = function(hostname, oldHostname, uid) {
+        var self = this;
+        RequestHandler.redirectHosts[hostname] = uid;
+        RequestHandler.sites[oldHostname] = null;
+
+        var dao = new pb.DAO();
+        // If a 301 loop is detected, delete needless redirect
+        if (RequestHandler.redirectHosts[oldHostname]) {
+            self.deleteRedirectHost(oldHostname);
+        }
+
+        var redirect = {host: hostname,
+                        uid: uid};
+        var redirectHost = pb.DocumentCreator.create('redirect_hosts', redirect);
+        dao.save(redirectHost, function(err, result) {
+            if (pb.util.isError(err)) {
+                pb.log.error("ERROR: Failed to save off redirectHost ERR[" + err.stack + "]");
+
+            }
+        });
+    };
+
+    /**
+     * @static
+     * @method deleteRedirectHost
+     * @param {String} hostname
+     */
+    RequestHandler.deleteRedirectHost = function(hostname) {
+        var dao = new pb.DAO();
+        dao.delete({host:hostname}, 'redirect_hosts', function(err, result) {
+            if (pb.util.isError(err)) {
+                pb.log.error("ERROR: Failed to delete deprecated redirectHost ERR[" + err.stack + "]");
+            }
+        });
+    };
+
+    /**
+     * @static
+     * @method checkForRedirect
+     * @param {String} hostname
+     * @param {Function} cb
+     */
+    RequestHandler.checkForRedirect = function(hostname, cb) {
+        var err = null,
+            siteObj = RequestHandler.sites[hostname];
+
+        if (siteObj) {
+            return cb(err, siteObj);
+        }
+
+        else {
+            var redirectSite = RequestHandler.redirectHosts[hostname];
+            if (redirectSite) {
+                new pb.SiteService().getAllSites(function (err, results) {
```

Comment:
I like it. I'll play around with this approach tomorrow. Thanks for the feedback!